CesiumIonRevitAddin/gltf/GltfExportContext.cs

```diff
@@ -452,14 +452,39 @@ public RenderNodeAction OnElementBegin(ElementId elementId)
                     }
                 }
 
-                // All nodes should have its ElementId in the metadata
-#if REVIT2022 || REVIT2023
-                int elementIdValue = elementId.IntegerValue;
-#else
-                int elementIdValue = (int)elementId.Value;
-#endif
+                // All nodes should have its ElementId, LevelId, and UniqueId in the metadata
+                int elementIdValue = Util.GetElementIdValue(elementId);
                 newNode.Extensions.EXT_structural_metadata.AddProperty("elementId", elementIdValue);
                 extStructuralMetadataExtensionSchema.AddSchemaProperty(categoryName, familyName, "elementId", elementIdValue.GetType());
+
+                string uniqueId = element.UniqueId;
+                newNode.Extensions.EXT_structural_metadata.AddProperty("uniqueId", uniqueId);
+                extStructuralMetadataExtensionSchema.AddSchemaProperty(categoryName, familyName, "uniqueId", uniqueId.GetType());
+
+                ElementId levelId = element.LevelId;
+                int levelIdValue = Util.GetElementIdValue(levelId);
+                newNode.Extensions.EXT_structural_metadata.AddProperty("levelId", levelIdValue);
+                extStructuralMetadataExtensionSchema.AddSchemaProperty(categoryName, familyName, "levelId", levelIdValue.GetType());
+                // add "Name" property for levelId for human readability
+                string levelIdName = Doc.GetElement(levelId).Name;
+                newNode.Extensions.EXT_structural_metadata.AddProperty("levelIdName", levelIdName);
+                extStructuralMetadataExtensionSchema.AddSchemaProperty(categoryName, familyName, "levelIdName", levelIdName.GetType());

```

Comment:
Snowdon Architecture export failed as an element was referencing a `levelId` of `-1`.  The name lookup obviously failed for this, so this changes it to only add `levelIdName` if it is a valid level

```suggestion
                if (levelElement != null)
                {
                    string levelIdName = levelElement.Name;
                    newNode.Extensions.EXT_structural_metadata.AddProperty("levelIdName", levelIdName);
                    extStructuralMetadataExtensionSchema.AddSchemaProperty(categoryName, familyName, "levelIdName", levelIdName.GetType());
                }

```